blog/2021-05-19-monero-transaction.md

```diff
@@ -0,0 +1,280 @@
+This is part of the ring signature, described in the next section.
+
+##### CLSAGs
+
+All that remains is proving ownership of the inputs we're spending.
+Once again, since we're spending one input out of a ring of possible candidates, we need something a bit more exotic than a regular Schnorr signature.
+The correct tool is a ring signature, a general term used to refer to signatures which prove that one member of a group of several parties approves of a message, without revealing the identity of the member.
+In our case, that is one input's secret key out of the ring of possible inputs authorising being spent in the transaction.
+
+CLSAG is one of several instantiations of this idea, and is the one currently used in Monero.
+Much like with Bulletproofs, it seems more appropriate to direct the reader elsewhere if they want to understand this signing scheme in more depth.
+In particular, we would like to recommend the excellent [_Zero to Monero_](https://www.getmonero.org/library/Zero-to-Monero-2-0-0.pdf), which has been an invaluable resource when conducting our research.
+Nonetheless, we would still like to mention all the particularities of Monero's implementation of CLSAG.
+
+###### Involving Pedersen commitments
+
+Vanilla CLSAG (as described in _Zero to Monero_) only concerns itself with authorising spending an input, but Monero's implementation has been modified to also confidentially prove that the real input used commits to the same amount as the pseudo output commitment included in the transaction.
+
+An iteration of the CLSAG algorithm commonly consists of computing a challenge of the form `c_i+1 = H_n(domain_tag || public_parameters || L_i+1 || R_i+1)`, where the domain tag and the public parameters remain constant between iterations, and `L` and `R` are expressions which depend on the previous iteration.
+We can therefore focus on the values of `L` and `R` and see how they're computed differently in Monero.
+
+Looking at `L`, regular CLSAG computes it as `L_i+1 = r_i * G + c_i * pk_i ` whereas Monero's instead does `L_i+1 = r_i * G + c_i * mu_P * pk_i + c_i * mu_C * (C_i - C_pseudo)`, where `mu_P` and `mu_C` are constants, `C_i` is the `i`th commitment in the ring, `r_i` is the `i`th response and `C_pseudo` is the pseudo output commitment.
+We can see that a third component has been introduced in Monero, which is kept orthogonal to the second one because `mu_P ≠ mu_C`.
+This new component is actually responsible for attesting the fact that the pseudo output commitment hides the same amount as the real input's commitment.
+
+Normally CLSAG enforces that to be able to _close_ the ring and generate the response `r_π` we need to know the real input's secret key `sk_π`: `r_π = ⍺ - c_π * sk_π`, where `⍺` is a random value generated at the start of the algorithm.
```

Comment:
Adding commas here might help the reader to make the mental pause required to correctly parse this sentence.

```suggestion
Normally CLSAG enforces that, to be able to _close_ the ring and generate the response `r_π`, we need to know the real input's secret key `sk_π`: `r_π = ⍺ - c_π * sk_π`, where `⍺` is a random value generated at the start of the algorithm.
```

Reply:
Agreed!